test/opentelemetry_tesla_test.exs

```diff
@@ -66,19 +46,33 @@ defmodule OpentelemetryTeslaTest do
 
     assert_receive {:span,
                     span(
-                      name: "External HTTP Request",
+                      name: "HTTP GET",
                       attributes: [
-                        {"http.method", :get},
-                        {"http.opts", []},
-                        {"http.query", []},
-                        {"http.status", nil},
-                        {"http.url", "http://end_of_the_inter.net"},
-                        {"http.request.measurement", 1000}
+                        "http.method": "GET",
+                        "http.url": "http://end_of_the_inter.net",
+                        "http.target": nil,
+                        "http.host": "end_of_the_inter.net",
+                        "http.scheme": "http",
+                        "http.status_code": nil
                       ]
                     )}
   end
 
   test "Records Spans for exceptions" do
+    tesla_env = %{
```

Comment:
Your tests should cover as much of the spec as possible. I suggest using a request that includes ports, paths, schemes, query strings, etc.

Reply:
Actually I'm going to refactor all tests, it doesn't make sense to me to be generating `%Tesla.Env{}` struct by my own.
Instead, I'm going to use the approach that Tesla uses. It relies on [httparrot](https://github.com/edgurgel/httparrot) to then use a Tesla client to make actual HTTP requests so that metadata is real and always up to data with the one that Tesla library uses.

Reply:
have a look at the bypass library

Reply:
> have a look at the bypass library

just pushed some changes using `bypass`, thanks for the tip 🙏

Reply:
I see, nice! Yeah I really like the `bypass` library, I used to always mock behaviors using `mox` but now I often can get away by just using the real deal with `bypass`.

Reply:
awesome 🙏 
if you like/can please take a look at the PR.